### PR TITLE
Fixed issue #1

### PR DESCRIPTION
**Pull request by iotanbo**

### CODE REVIEW

**Comment by DavyLandman** [comment]
Thank you, there are two things i would like to note:

- I'm not a fan of copying the private key somewhere the caller is not aware of. It should be cleared after use.
- how about a small rfc7748 variant to generate public private keys?


Could you merge from master? I've update the Github Actions config to also run on PRs.

**Comment by DavyLandman** [comment]
Hmm, having spent a bit more time with the code again.

I think my design was that you should not be feeding in raw private keys, but use the gen function to generate private an public keys.

**Comment by iotanbo** [comment]
I've updated the PR to include your recent commit.
As for the design decision not to feed raw secret keys, that's totally fine IMHO. But there may be a use case when for whatever reason  someone wants to provide his own private key. Another use case is to test against RFC test vectors, which are raw.
As for my fix, it just makes sure that even in this case the secret key will be clamped as RFC suggests. In no way it will modify already clamped secret key.

**Comment by DavyLandman** [comment]
I'm conflicted, I understand where you are coming from, but it's also just wrong usage of the API.

I can understand adding this safety feature, as it would prevent users from incorrectly calling the function with a raw key (how they got it, is a bit of a mess, but okay). 

I've added a comment about wiping the memory.

File: src/compact_x25519.c [hunk 1]
```diff
@@ -21,7 +21,12 @@ void compact_x25519_shared(
     const uint8_t my_private_key[X25519_KEY_SIZE], 
     const uint8_t their_public_key[X25519_KEY_SIZE]
 ) {
-    c25519_smult(shared_secret, their_public_key, my_private_key);
+    // Ensure that supplied private key is clamped (fix issue #1).
+    // Applying `c25519_prepare` multiple times to same private key is fine.
+    uint8_t clamped_private_key[X25519_KEY_SIZE];
```
Comment (DavyLandman) [inline src/compact_x25519.c:27]: this array now contains the private key, we don't want to leave this lying around outside of the control of the caller.

It should be `compact_wipe` after usage.

Comment (iotanbo) [reply src/compact_x25519.c:27]: That's reasonable. Applied.

**Review by DavyLandman — APPROVED** [review]

**Pull request merged**